crates/ty_python_semantic/src/types.rs

```diff
@@ -1419,6 +1432,16 @@ impl<'db> Type<'db> {
             // All `StringLiteral` types are a subtype of `LiteralString`.
             (Type::StringLiteral(_), Type::LiteralString) => true,
 
+            // An instance is a subtype of an enum literal, if it is an instance of the enum class
+            // and the enum has only one member.
+            (Type::NominalInstance(self_instance), Type::EnumLiteral(target_enum_literal)) => {
+                if target_enum_literal.enum_class_instance(db) != self {
+                    return false;
+                }
+
+                is_single_member_enum(db, self_instance.class.class_literal(db).0)
```

Comment:
this feels like slightly less indirection (probably won't make much of a difference for performance, but maybe slightly more readable?)

```suggestion
                is_single_member_enum(db, target_enum_literal.enum_class(db))
```